### DISCUSSION TITLE
Change the allocation for the EEPROM and the program memory

### DISCUSSION THREAD

**Discussion by taltalp** [stm32duino/Arduino_Core_STM32#1957] [Q&A]
Hello,

We are trying to use the EEPROM library for logging temperature, and we use STM32L452RE which has a 512kB flash.
As we read the [EEPROM library](https://github.com/stm32duino/Arduino_Core_STM32/blob/8ae77249fe04ce4f693c57b9244a465b4b82529d/libraries/EEPROM/src/EEPROM.h), its size is [limited up to 2kB](https://github.com/stm32duino/Arduino_Core_STM32/blob/8ae77249fe04ce4f693c57b9244a465b4b82529d/libraries/EEPROM/src/EEPROM.h#L223-L226), however the program memory has 510kB.
Can we change the allocation for that?
And where is the memory map defined?

**Comment by fpistm** [comment]
Hi, I really not recommend to use EEPROM to log data. It uses the flash memory which has a limited number of R/W cycles.

**Reply by taltalp** [reply]
Thank you for your quick reply. I understand that, but I'm still looking for the way to expand the user memory size.

**Reply by fpistm** [reply]
If you don't use the EEPROM library all flash is used. Flash is not reserved for it.

Example for Nucleo L452RE:
https://github.com/stm32duino/Arduino_Core_STM32/blob/621d75921205acdeef53220897c6f815c227a3f0/boards.txt#LL563C46-L563C47

and its linker script:
https://github.com/stm32duino/Arduino_Core_STM32/blob/621d75921205acdeef53220897c6f815c227a3f0/variants/STM32L4xx/L452RETxP/ldscript.ld#L40

`LD_FLASH_OFFSET` is 0 so the `LD_MAX_SIZE` is set to max size defined in the boards.txt: 524288 (512K).

**Reply by taltalp** [reply]
Thank you. 
So if we don't overwrite the firmware on the flash, the full addresses are open to write?
Or is there any write protections for the program area?

**Reply by fpistm** [reply]
You have access to all the flash. Protection is generally managed by option bytes which is out of scope of stm32duino. It is managed by user.

**Reply by taltalp** [reply]
Ok that's make sense.
Thank you for all your support!

**Reply by fpistm** [reply]
Welcome 😉
